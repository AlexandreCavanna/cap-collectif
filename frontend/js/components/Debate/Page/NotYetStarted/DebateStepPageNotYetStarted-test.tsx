/* eslint-env jest */
import * as React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import { graphql, useLazyLoadQuery } from 'react-relay'
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils'
import { RelaySuspensFragmentTest } from '~/testUtils'
import type { DebateStepPageNotYetStartedTestQuery } from '~relay/DebateStepPageNotYetStartedTestQuery.graphql'
import DebateStepPageNotYetStarted from './DebateStepPageNotYetStarted'
describe('<DebateStepPageNotYetStarted />', () => {
  let environment
  let testComponentTree
  let TestComponent
  const query = graphql`
    query DebateStepPageNotYetStartedTestQuery($id: ID = "<default>") @relay_test_operation {
      step: node(id: $id) {
        ...DebateStepPageNotYetStarted_step
      }
    }
  `
  beforeEach(() => {
    environment = createMockEnvironment()

    const TestRenderer = () => {
      const data = useLazyLoadQuery<DebateStepPageNotYetStartedTestQuery>(query, {})
      if (!data.step) return null
      return <DebateStepPageNotYetStarted step={data.step} />
    }

    TestComponent = () => (
      <RelaySuspensFragmentTest environment={environment}>
        <TestRenderer />
      </RelaySuspensFragmentTest>
    )
  })
  it('should render when step is future', async () => {
    ;(Date as any).now = jest.fn(() => 1517848349000) // 5/02/2018

    const openStepMockResolvers = {
      DebateStep: () => ({
        timeRange: {
          startAt: '2020-02-01 00:03:00',
        },
      }),
    }
    environment.mock.queueOperationResolver(operation =>
      MockPayloadGenerator.generate(operation, openStepMockResolvers),
    )
    testComponentTree = ReactTestRenderer.create(<TestComponent />)
    expect(testComponentTree).toMatchSnapshot()
  })
})
