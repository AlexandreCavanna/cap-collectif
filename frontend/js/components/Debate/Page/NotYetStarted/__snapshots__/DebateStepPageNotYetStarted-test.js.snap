// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DebateStepPageNotYetStarted /> should render when step is future 1`] = `
<div
  className="sc-bdnxRM"
  id="DebateStepPageNotYetStarted"
>
  <div
    className="sc-bdnxRM cefOPV"
  >
    <div
      className="sc-bdnxRM zqsiY sc-fXazdy hCrgT cap-tag"
      title=""
    >
      count.daysUntil {"count":725}
    </div>
    <h2
      className="sc-bdnxRM kYuusp cap-heading"
    >
      &lt;mock-value-for-field-"title"&gt;
    </h2>
  </div>
</div>
`;
